regen-network/regen-web
# perf: Lazy load routes

## Description

Closes: regen-network/regen-registry#992

All pages are now lazy-loaded and each page has its own bundle. The main bundle is down by **~50KB**
For now this only amount for application code, webpack is not able to split the vendor bundle yet, it would need more tweaking.

__Edit:__
I have just pushed a new commit that remove mapbox dependency from the main bundle, it saves us an additional  **224.25 KB** from the main vendor bundle !

__Edit2:__
Save another **30KB** by splitting rdf validation.

| Before  | After  |
|---|---|
| ![before](https://camo.githubusercontent.com/83d41b0acda09846d6837330b4d6eea9a6286fa71f8f38daf5d30cdf44772b05/68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3632613937643337393630336561653039303336663862622f38336462383134632d343430302d346434312d616636322d663566393933333938366435)  |  ![after](https://user-images.githubusercontent.com/530644/176711006-e5b9d7c6-9a14-42c5-bbb4-f4a17fd2e6db.png) |


Also in order to Lazy-load pages, we needed to have a default export for all pages. As it was discussed with @haveanicedavid [here](https://github.com/regen-network/regen-web/pull/1003#discussion_r907656023) we should avoid using default export in the component file directly (see also main react doc [here](https://reactjs.org/docs/code-splitting.html#route-based-code-splitting) about this). In order to do that all pages have now been moved to their own folder with an `index.ts` file which does the actual default export.

![folder architecture](https://user-images.githubusercontent.com/530644/176667502-18c0b5be-87e4-49a4-88a4-07f7463ebe8a.png)

## How to review this PR?
- check `App.tsx` changes
- check that pages that have been moved only have import changes
- Go the [preview URL](https://deploy-preview-1017--regen-registry.netlify.app/) and check that pages are lazy-loaded while navigating (you might see a loader or you can check the network tab).

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...


#### Setting up backport PRs

After merging your PR to `master`, set up backports by doing the following:

1. If your branch does not have merge commits, add the following comment to
   your PR, `@Mergifyio backport redwood hambach`.

2. If your branch does have merge commits:

    a. Pull latest `master`, `hambach` and `redwood` branches

    b. Create new branches for backports and merge `master` (replace `<PR#>` with your PR #)
    ```
    git checkout -b hambach-backport-<PR#> hambach
    git merge master
    git push origin hambach-backport-<PR#>
    git checkout -b redwood-backport-<PR#> redwood
    git merge master
    git push origin redwood-backport-<PR#>`
    ```

    c. Open new PRs in regen-web targeting `hambach` and `redwood`, respectively.
  

### How to test

1.

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**._

I have...
